cli/aws_orbit/remote_files/cdk/env.py

```diff
@@ -223,15 +223,20 @@ def _get_user_pool(self, user_pool_arn: str) -> cognito.UserPool:
             cognito.UserPool.from_user_pool_arn(scope=self, id="orbit-user-pool", user_pool_arn=user_pool_arn),
         )
 
-    def _create_user_pool_client(self) -> cognito.UserPoolClient:
-        return cognito.UserPoolClient(
+    def _create_user_pool_client(self) -> cognito.CfnUserPoolClient:
```

Comment:
So I ran into an issue where `mypy aws_orbit` was failing for adding both `access_token_validity` and `refresh_token_validity` to the L2 UserPoolClient construct.

@stthoom was very helpful in pointing out that it might be better to just the L1 CfnUserPoolClient instead, to ensure we can set our custom expiry needs for the token.

After digging in, stable support for `UserPoolClient` wasn't introduced until aws_cdk version 1.77, shown here:
[1.77 release](https://github.com/aws/aws-cdk/releases/tag/v1.77.0)

In my opinion, an overall solution should be to version bump cdk to at least that, if not even higher to get better support overall for orbit in the future.  

That being said, I don't feel too comfortable doing a version bump since we don't currently have great coverage cdk validation testing.  Each cdk release mentions breaking changes along the way.

For now, I've swapped to use a L1 construct, but needs to be revisited at some point.  Open to suggestions or thoughts on this matter.  Can also capture this as a separate issue to be addressed.

Reply:
@chamcca

Reply:
we lock to a specific version of the CDK because they release frequently and don't announce breaking changes before release. i've had too many projects start to fail after a new CDK release..

right now we're locked to 1.67 as upgrading requires a rebuild of and full regression testing on our cli, sdk, plugins, as well as the custom codebuild docker image we build. this isn't currently in the backlog.

do you have a hard requirement for the L2 construct right now? i know the L2s are recommended and easier to use, but updating the CDK version is non-trivial.

Reply:
Nope, and I agree updating CDK versions are non-trivial.  Was just a nice to have, so wanted to be explicit in why I'm switching the construct here.